### DISCUSSION TITLE
Change useBlockNumber to useBlock

### DISCUSSION THREAD

**Discussion by nasa8x** [wevm/wagmi#673] [Ideas]
Providing blockNumber information almost has little benefit, instead providing full information of a full block (ex: timestamp, number, hash...) in one feature will be much more useful.

Do you have the same thoughts as me?

**Comment by wulimao49** [comment]
agree, a config to tell what we need to fetch is needed

**Comment by jxom** [comment]
A separate `useBlock` hook that wraps the ethers [providers.getBlock](https://docs.ethers.io/v5/api/providers/provider/#Provider-getBlock) could be feasible (and still keep `useBlockNumber`)

**Reply by SpasiboKojima** [reply]
Any updates or plans on this?

**Comment by TheDojoSamurai** [comment]
Agreed a `useBlock` would be very helpful. 
I need to grab timestamp of current block

**Comment by jxom** [comment]
We are going to be working on this v soon 😘

**Comment by TheDojoSamurai** [comment]
Great to hear. In the meantime I used this

`const blockInfo = await provider.getBlock(blockNumber)`

**Comment by anaszaman** [comment]
If someone looking for complete code, here it is;
You can use custom hook to get timestamp until wagmi make it official.
```
import { usePublicClient } from 'wagmi'
import { useEffect, useState } from 'react';

export function useBlock() {
    const client = usePublicClient();
    const [block, setBlock] = useState<any>(null);

    useEffect(() => {
        async function fetchBlock() {
            try {
                const block = await client.getBlock();
                setBlock(block);
            } catch (error) {
                console.error('Error fetching block:', error);
            }
        }

        fetchBlock();
    }, [client]);

    return block
}
```
